lib/ansible/plugins/connection/ssh.py

```diff
@@ -344,10 +349,19 @@ def wrapped(self, *args, **kwargs):
                     display.vvv(u"RETRYING BECAUSE OF CONTROLPERSIST BROKEN PIPE")
                     return_tuple = func(self, *args, **kwargs)
 
-                if return_tuple[0] != 255:
-                    break
-                else:
-                    raise AnsibleConnectionFailure("Failed to connect to the host via ssh: %s" % to_native(return_tuple[2]))
+                if return_tuple[0] == 255:
+                    SSH_ERROR = True
+                    for signature in NOT_SSH_ERRORS:
+                        if signature in return_tuple[1]:
+                            SSH_ERROR = False
```

Comment:
You can raise from here.

Reply:
I don't see how.  Could you explain?

Reply:
Oh.. I swapped the logic in my mind. Yeah, no way w/o `else:`.
(I still think that semantically `all()` reads more obvious)